addons/website_sale_autocomplete/models/autocompletion_provider.py

```diff
@@ -0,0 +1,145 @@
+                         ('country_id.id', '=', standard_data['country'])])
+                    if len(state) == 1:
+                        standard_data[field_standard] = state.id
+                else:
+                    standard_data[field_standard] = google_field['long_name']
+        return standard_data
+
+    def _guess_number_from_input(self, source_input, standard_address):
+        """
+        Google might not send the house number in case the address
+        does not exist in their database.
+        We try to guess the number from the user's input to avoid losing the info.
+        """
+        # Remove other parts from address to make better guesses
+        guess = source_input.replace(standard_address.get('zip', ''), '') \
+            .replace(standard_address.get('street', ''), '') \
+            .replace(standard_address.get('city', ''), '')
+        guess = guess.split(',')[0].strip()
+        return guess
+
+    def _autocomplete_full(self, address, google_place_id=None, language_code=None, session_uuid=None):
+        params = {
+            'key': self._get_key(),
+            'place_id': google_place_id,
+            'fields': 'address_component,adr_address'
+        }
+
+        if language_code:
+            params['language'] = language_code
+        if session_uuid:
+            params['sessiontoken'] = session_uuid
+
+        results = requests.get(f'{GOOGLE_PLACES_ENDPOINT}/details/json', params=params).json()
+
+        html_address = results['result']['adr_address']
+        results = results['result']['address_components']  # Get rid of useless extra data
+
+        # Keep only the first type from the list of types
+        for res in results:
+            res['type'] = res.pop('types')[0]
+
+        # Sort the result by their priority.
+        results.sort(key=lambda r: FIELDS_PRIORITY.index(r['type']) if r['type'] in FIELDS_PRIORITY else 100)
+
+        standard_address = self._translate_google_to_standard(results)
+
+        if 'number' not in standard_address:
+            standard_address['number'] = self._guess_number_from_input(address, standard_address)
+            standard_address['formatted_street_number'] = f'{standard_address["number"]} {standard_address["street"]}'
+        else:
+            formatted_from_html = html2plaintext(html_address.split(',')[0])
+            formatted_manually = f'{standard_address["number"]} {standard_address["street"]}'
+            # Sometimes, the google api sends back abbreviated data :
+            # "52 High Road Street" becomes "52 HR St" for example. We usually take the result from google, but if it's an abbreviation, take our guess instead.
+            if len(formatted_from_html) >= len(formatted_manually):
+                standard_address['formatted_street_number'] = formatted_from_html
+            else:
+                standard_address['formatted_street_number'] = formatted_manually
```

Comment:
Be careful of `KeyError`s it will crash if `'street'` is not in `standard_address`; can this happen?
I would guard those just in case.